components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSOAPUtils.java

```diff
@@ -0,0 +1,281 @@
+     * @param authReqStr Authentication Request
+     * @return
+     * @throws IdentityException
+     * @throws IdentitySAML2ECPException
+     */
+    public static XMLObject unmarshall(String authReqStr) throws IdentityException, IdentitySAML2ECPException {
+        InputStream inputStream = null;
+        doBootstrap();
+        try {
+            DocumentBuilderFactory documentBuilderFactory = IdentityUtil.getSecuredDocumentBuilderFactory();
+            DocumentBuilder docBuilder = documentBuilderFactory.newDocumentBuilder();
+            inputStream = new ByteArrayInputStream(authReqStr.trim().getBytes(StandardCharsets.UTF_8));
+            Document document = docBuilder.parse(inputStream);
+            Element element = document.getDocumentElement();
+            UnmarshallerFactory unmarshallerFactory = Configuration.getUnmarshallerFactory();
+            Unmarshaller unmarshaller = unmarshallerFactory.getUnmarshaller(element);
+            return unmarshaller.unmarshall(element);
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            String err = "Error in Processing the  SAML request";
+            throw new IdentitySAML2ECPException(err);
+        }
+    }
+
+    /**
+     *
+     * Decode the request recived by the /ecp servlet.
+     * Validate the SOAP message
+     * Check whether the SOAP body contains a valid SAML request
+     * @param soapMessage
+     * @return
+     * @throws IdentitySAML2ECPException
+     */
+    public static String decodeSOAPMessage(SOAPMessage soapMessage) throws IdentitySAML2ECPException {
+        SOAPBody body = null;
+        String samlRequest = null;
+        String message = null;
+        try {
+            body = soapMessage.getSOAPPart().getEnvelope().getBody();
+        } catch (SOAPException e) {
+            log.error("Error Processing the SOAP Mesage");
+            throw new IdentitySAML2ECPException(e.getMessage());
+        }
+
+        int elementSize = 0;
+        try {
+            Iterator<?> elements = body.getChildElements();
+            while (elements.hasNext()) {
+                SOAPElement element = (SOAPElement) elements.next();
+                DOMSource source = new DOMSource(element);
+                StringWriter stringResult = new StringWriter();
+                try {
+                    TransformerFactory.newInstance().newTransformer().transform(source, new StreamResult(stringResult));
+                } catch (TransformerException e) {
+                    String err = "Transformer Exception";
+                    log.error(err);
+                    throw new IdentitySAML2ECPException(e.getMessage());
+                }
+                message = stringResult.toString().replace("<?xml version=\"1.0\" encoding=\"UTF-8\"?>", "");
+                samlRequest = Base64.getEncoder().encodeToString(message.getBytes());
+                elementSize += 1;
+            }
+        } catch (NullPointerException e) {
```

Comment:
We usually don't catch runtime exceptions such as NPE. Any specific reason to do this ?

Reply:
> We usually don't catch runtime exceptions such as NPE. Any specific reason to do this 
removed the exception